### PR TITLE
refactor(bg/storage): use separate key to toggle continuous payments

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

Will help with https://github.com/interledger/web-monetization-extension/issues/623 as decided in https://github.com/interledger/web-monetization-extension/pull/654#issuecomment-2407142584

Then in future `enabled` will mean entire extension enabled or not, not just continuous payments.

## Changes proposed in this pull request

- Rename `Storage.enabled` to `Storage.continuousPaymentsEnabled`
- Migrate extension storage from version 3 to 4
  - set `continuousPaymentsEnabled` to previous value of `enabled`
  - set `enabled` to `true` for [#623](https://github.com/interledger/web-monetization-extension/issues/623)
- Update entire codebase to reflect this rename

## Notes to reviewer

- A shorter name suggestion is definitely welcome.

### CODE REVIEW

File: src/background/services/storage.ts [hunk 1]
```diff
@@ -284,4 +286,9 @@ const MIGRATIONS: Record<Storage['version'], Migration> = {
     data.state = newState satisfies Storage['state'];
     return [data];
   },
+  4: (data) => {
+    data.continuousPaymentsEnabled = data.enabled === true;
```
Comment (raducristianpopa) [inline src/background/services/storage.ts:290]:
```suggestion
    data.continuousPaymentsEnabled = data.enabled
```

**Review by raducristianpopa — COMMENTED** [review]
Now thinking about it, we might not even need to delete the `enabled` key now, if we plan to introduce the global `enabled` flag. We can just keep it for when that lands.

**Comment by sidvishnoi** [comment]
Yes, but its value isn't same as current meaning of enabled. We'll add it back (as new migration) with its new meaning.

**Comment by sidvishnoi** [comment]
Update: oh yeah we can keep it. Just not use its value or set it yet.

**Comment by DarianM** [comment]
LGTM

**Pull request merged**